index.js

```diff
@@ -27,12 +29,22 @@ function NYC (opts) {
   this._istanbul = opts.istanbul
   this.subprocessBin = opts.subprocessBin || path.resolve(__dirname, './bin/nyc.js')
   this._tempDirectory = opts.tempDirectory || './.nyc_output'
-  this.cwd = opts.cwd || process.env.NYC_CWD || process.cwd()
-  this.reporter = arrify(opts.reporter || 'text')
+
+  var cwd = opts.cwd || process.env.NYC_CWD || process.cwd()
+  var pkgPath = pkgUp.sync(cwd)
 
   // you can specify config in the nyc stanza of package.json.
-  var config = require(path.resolve(this.cwd, './package.json')).config || {}
-  config = config.nyc || {}
+  var config
+  if (pkgPath) {
+    cwd = path.dirname(pkgPath)
+    var pkg = readPkg.sync(pkgPath, {normalize: false})
+    config = pkg.nyc || (pkg.config && pkg.config.nyc)
+  }
+  config = config || {}
+
+  this.cwd = cwd
```

Comment:
what if we have a library under coverage that has a few inner `package.json` files? perhaps in a fixtures directory, or with some projects like [babel](https://github.com/babel/babel/tree/master/packages) I've seen a mono-repo approach that has many package.json files, is there a danger that we'll accidentally source the inner package.json in a subprocess rather than the root directory's?